unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -3319,3 +3327,45 @@ fuzzySelectNamespace searchBranch0 =
           Path.toText
           (Set.toList $ Branch.deepPaths searchBranch0)
       )
+
+
+-- | Check the dependencies of all types, terms, and metadata in the current namespace,
+-- returns dependencies which do not have a name within the current namespace, alongside their
+-- type (for terms and constructors).
+--
+-- This is non-transitive, i.e. only the first layer of external dependencies is returned.
+--
+-- So if my namespace depends on .base.Bag.map; which depends on base.Map.mapKeys, only
+-- .base.Bag.map is returned unless some other definition inside my namespace depends
+-- on .base.Bag.map directly.
+namespaceDependencies :: forall m i v. Branch0 m -> Action m i v (Map Reference (Type v Ann), Map Reference (Set Referent))
+namespaceDependencies branch = do
```

Comment:
@aryairani  Just want a sanity check on this function when you've got a second 😄

Reply:
method (3)? 🤔 

Not sure what you're referring to here 😓

Reply:
Method (3) (from https://github.com/unisonweb/unison/issues/1802#issuecomment-957938180) is when we migrate the codebase in the future to expose more details relationally.